### DISCUSSION TITLE
How to link against newer versions of libffi?

### DISCUSSION THREAD

**Discussion by fyrak1s** [ImageMagick/ImageMagick#7729] [Development]
### ImageMagick version

 7.1.1.39

### Operating system, version and so on

GoboLinux

### Description

Hi,

I am compiling ImageMagick 7.1.1.39 with Libffi v3.4.6 installed, which provides `libffi.so.8`.

However upon running `convert`, I get:
```
convert: error while loading shared libraries: libffi.so.7: cannot open shared object file: No such file or directory
```
If I install an older version of Libffi which has `libffi.so.7` (eg 3.3) the tool runs fine.

How do I build ImageMagick against the newer libffi?
Or does ImageMagick strictly require `libffi.so.7`? Would I be better of doing a compatibility symlink to of `libffi.so.7` to `libffi.so.8`?

**Comment by fmw42** [comment]
Note that with IM 7, you should use magick rather than convert

**Reply by fyrak1s** [reply]
Thanks for making me aware; although in this case it yields the same result.
